### ISSUE TITLE
freqVector Domain/default pair non-existent

### ISSUE THREAD

**Issue by rajesheber** [Piker-Alpha/freqVectorsEdit.sh#5]
I tried to run the freqVectorsedit.sh to edit my Mac-E43C1C25D4880AD6.plist (MacBookPro12,1) to change the LTF value from 1300Ghz to 600 mhz as per intel website but am getting the following error. Kindly help me resolve this so that i can run ssdtgenerator.sh to correct this issue.

```
2016-04-02 10:39:29.836 defaults[2321:42299] 
The domain/default pair of (/Users/raj/Library/Preferences/com.wordpress.pikeralpha, freqVectorsEditor) does not exist
```

**Comment by Piker-Alpha** [comment]
It failed to read a preference. Something I hope to have resoled in v2.0 so please give it a go and let me know if that is the case.

**Comment by rajesheber** [comment]
thank for the response. i ran the v2.0 and now when i run the ssdt generator i still get this error
Notice: The LFM frequency in Mac-E43C1C25D4880AD6.plist is set to 1300MHz!
    This problem can be fixed with help of freqVectorsEdit.sh from:
    https://github.com/Piker-Alpha/freqVectorsEdit.sh

**Comment by Piker-Alpha** [comment]
Correct. The current implementation of this 'check' will always show the 'notice'. Even when you patched it in the FrequencyVectors. The problem is that we don't check the data, but make an assumption based on the board-id. This is something on my todo list, but there is plenty other work t do as well. Will keep you posted here!

Thank you for bringing this up here!

**Comment by r2tincan** [comment]
I don't have the option for MacPro6,1 when I run the script. What am I doing wrong?

**Comment by Piker-Alpha** [comment]
@r2tincan,

The MacPro6,1 is the default.

**Comment by ajschot** [comment]
Ok MacPro6,1 is default.... we'll how can I get a good SSDT then? Because it gives me an error to use this tool for frequency. I now get my E5-2658v4 always on 2.3GHz never go into turbo nor lover speeds. I put the processor for ssdtgen in my user defined with the right values. then I need this tool for the right things, if MacPro6,1 is default how do I select it then? I need a number 1-30 or exit.
Tried to press nothing but this did not work.

**Comment by Piker-Alpha** [comment]
The MacPro6,1 is the default, but if you use any other model, then the plist of that model will be used and your **target** file.

The next step is to select a board-id.plist as your **source** file. The source file is used to copy the FrequencyVectors from. They will be added to the plist for the model that you are using.

**Edit**: There are no FrequencyVectors in the MacPro6,1 plist and selecting it would thus make it fail.

**Comment by raymanfx** [comment]
@ajschot The tool is indeed very confusing because it apparently 'omits' the default value from the list.
Here's what I did to fix it:

```
--- freqVectorsEdit_orig.sh	2017-09-17 11:45:07.000000000 +0200
+++ freqVectorsEdit.sh	2017-09-17 12:04:46.000000000 +0200
@@ -595,7 +595,8 @@
 {
   cd "${gResourcePath}"
 
-  gTargetFileNames=($(grep -rlse 'FrequencyVectors' .))
+  gTargetFileNames="Mac-F60DEB81FF30ACF6.plist"
+  gTargetFileNames+=($(grep -rlse 'FrequencyVectors' .))
 
   if [[ "${#gTargetFileNames[@]}" -eq 0 ]];
     then
```

@Piker-Alpha by the way, the _listmatchingFiles() func is missing in the latest version of your script. It's supposed to be called when you select a list index of <= 0 or > $index.

**Comment by Piker-Alpha** [comment]
@raymanfx,

**1.**) I don't see why you want to do make that change. The factory plist for the MacPro6,1 comes without FrequencyVectors and that is the only reason it won't show up in the list.

**2.**) _listmatchingFiles() should have been named _selectSourceResourceFile() but I guess I forgot to change that. Thank you for noticing it and reporting it here.  

**Fixed in v3.2**
